File: app/src/main/java/com/fruitblast/game/MainActivity.java
package com.fruitblast.game;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.fruitblast.game.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private PreferenceHelper prefs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        prefs = new PreferenceHelper(this);

        binding.btnEasy.setOnClickListener(v -> launchGame(Difficulty.EASY));
        binding.btnMedium.setOnClickListener(v -> launchGame(Difficulty.MEDIUM));
        binding.btnHard.setOnClickListener(v -> launchGame(Difficulty.HARD));
        binding.btnHighScores.setOnClickListener(v ->
                startActivity(new Intent(this, HighScoreActivity.class)));
        binding.btnSound.setOnClickListener(v -> { /* reserved for sound toggle */ });
    }

    @Override
    protected void onResume() {
        super.onResume();
        refreshBestScores();
    }

    private void refreshBestScores() {
        int easy   = prefs.getBestScore(Difficulty.EASY);
        int medium = prefs.getBestScore(Difficulty.MEDIUM);
        int hard   = prefs.getBestScore(Difficulty.HARD);

        binding.tvEasyBest.setText(easy > 0
                ? getString(R.string.easy_desc) + "  |  最高: " + easy
                : getString(R.string.easy_desc));
        binding.tvMediumBest.setText(medium > 0
                ? getString(R.string.medium_desc) + "  |  最高: " + medium
                : getString(R.string.medium_desc));
        binding.tvHardBest.setText(hard > 0
                ? getString(R.string.hard_desc) + "  |  最高: " + hard
                : getString(R.string.hard_desc));
    }

    private void launchGame(Difficulty difficulty) {
        Intent intent = new Intent(this, GameActivity.class);
        intent.putExtra(GameActivity.EXTRA_DIFFICULTY, difficulty.name());
        startActivity(intent);
    }
}

File: app/src/main/java/com/fruitblast/game/ui/MainActivity.kt
package com.fruitblast.game.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import com.fruitblast.game.R
import com.fruitblast.game.data.Difficulty
import com.fruitblast.game.databinding.ActivityMainBinding
import com.fruitblast.game.utils.PreferenceHelper

/**
 * Main menu screen.
 * Shows app title, difficulty selection buttons, and high scores entry.
 */
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var prefs: PreferenceHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        prefs = PreferenceHelper(this)

        setupUI()
        animateEntrance()
    }

    private fun setupUI() {
        binding.btnEasy.setOnClickListener { launchGame(Difficulty.EASY) }
        binding.btnMedium.setOnClickListener { launchGame(Difficulty.MEDIUM) }
        binding.btnHard.setOnClickListener { launchGame(Difficulty.HARD) }
        binding.btnHighScores.setOnClickListener {
            startActivity(Intent(this, HighScoreActivity::class.java))
        }

        // Show best scores for each difficulty as subtitles
        updateBestScores()

        // Sound toggle
        updateSoundButton()
        binding.btnSound.setOnClickListener {
            prefs.soundEnabled = !prefs.soundEnabled
            updateSoundButton()
        }
    }

    private fun updateBestScores() {
        val easyBest = prefs.getBestScore(Difficulty.EASY)
        val medBest = prefs.getBestScore(Difficulty.MEDIUM)
        val hardBest = prefs.getBestScore(Difficulty.HARD)

        if (easyBest > 0) binding.tvEasyBest.text = getString(R.string.best_score, easyBest)
        if (medBest > 0)  binding.tvMediumBest.text = getString(R.string.best_score, medBest)
        if (hardBest > 0) binding.tvHardBest.text = getString(R.string.best_score, hardBest)
    }

    private fun updateSoundButton() {
        binding.btnSound.text = if (prefs.soundEnabled) "🔊" else "🔇"
    }

    private fun launchGame(difficulty: Difficulty) {
        val intent = Intent(this, GameActivity::class.java)
        intent.putExtra(GameActivity.EXTRA_DIFFICULTY, difficulty.name)
        startActivity(intent)
        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
    }

    private fun animateEntrance() {
        val slideUp = AnimationUtils.loadAnimation(this, android.R.anim.slide_in_left)
        binding.titleLayout.startAnimation(slideUp)
    }

    override fun onResume() {
        super.onResume()
        updateBestScores()
    }
}
